solana-programs/anchor/audius-data/programs/audius-data/src/lib.rs

```diff
@@ -442,6 +471,49 @@ pub struct UpdateAdmin<'info> {
     pub admin_authority: Signer<'info>,
 }
 
+/// Instruction container to intialize an AppDelegation.
+/// The authority initializes itself as a delegate.
+/// `delegate_authority` is the authority that will become a delegate
+/// `app_delegation_pda` is the target PDA for the authority's delegation
+#[derive(Accounts)]
+pub struct InitAppDelegation<'info> {
+    /// CHECK: Delegate authority account
+    #[account()]
+    pub delegate_authority: Signer<'info>,
+    #[account(
+        init,
+        payer = payer,
+        seeds = [APP_DELEGATION_SEED, delegate_authority.key().as_ref()],
+        bump,
+        space = APP_DELEGATION_ACCOUNT_SIZE
+    )]
+    pub app_delegation_pda: Account<'info, AppDelegation>,
+    #[account(mut)]
+    pub payer: Signer<'info>,
+    pub system_program: Program<'info, System>,
+}
+
+/// Instruction container to revoke an AppDelegation.
+/// The authority revokes itself as a delegate.
+/// `delegate_authority` is the authority that will become a delegate
+/// `app_delegation_pda` is the target PDA for the authority's delegation
+#[derive(Accounts)]
+#[instruction(app_delegation_bump:u8)]
+pub struct RevokeAppDelegation<'info> {
+    /// CHECK: Delegate authority account
+    #[account()]
+    pub delegate_authority: Signer<'info>,
+    #[account(
```

Comment:
@raymondjacobson - I think instead of `is_revoked` we can just close the account and check the lamports balance. wdyt?

Reply:
can we ensure that operation is only 1 way though? i dont think so

Reply:
yes I agree here let's just keep it as is_revoked so it's one way only - definitely makes more sense